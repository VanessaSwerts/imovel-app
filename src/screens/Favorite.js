import React, { useEffect, useState } from 'react'
import { KeyboardAvoidingView, View, StyleSheet, Text, SafeAreaView, FlatList, TouchableWithoutFeedback } from 'react-native'
import { ImovelCard } from '../components'
import { useNavigation } from '@react-navigation/native'
import AsyncStorage from '@react-native-community/async-storage'
import api from '../services/api'

import colors from '../constants/colors.json'

export default function Favorite() {
  const navigation = useNavigation()

  const [favorites, setFavorites] = useState([])

  const getFavorites = async () => {
    const token = await AsyncStorage.getItem("user-token")
    if (!token) {
      navigation.reset({
        index: 0,
        routes: [{ name: "SignIn" }],
      })
      return
    }

    const config = {
      headers: {
        "Authorization": "Bearer " + token
      }
    }

    api.get(`/user/favorites`, config)
      .then((res) => {
        setFavorites(res.data)
      })
      .catch((err) => {
        console.error(err)
      })
  }

  useEffect(() => {
    getFavorites()
  }, [])

  return (
    <KeyboardAvoidingView style={styles.container} >
      <Text numberOfLiner={2} style={styles.title} >Favoritos</Text>


      <SafeAreaView style={styles.listContainer}>
        <FlatList
          data={favorites}
          keyExtractor={item => item.id.toString()}
          showsVerticalScrollIndicator={false}
          renderItem={({ item }) => {
            return (
              <TouchableWithoutFeedback
                onPress={() => navigation.navigate('IMovelDetails', { item })}
              >
                <View>
                  <ImovelCard item={item} colorIcon={'red'} />
                </View>
              </TouchableWithoutFeedback>
            )
          }}
        />
      </SafeAreaView>

    </KeyboardAvoidingView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors["platinum"],
    paddingVertical: 10,
    paddingHorizontal: 30,
  },

  title: {
    margin: 10,
    fontSize: 28,
    fontWeight: "600",
    color: colors['yellow'],
    alignSelf: 'center',
  },

  listContainer: {
    flex: 1
  }

})