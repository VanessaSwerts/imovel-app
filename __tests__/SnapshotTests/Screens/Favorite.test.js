import React from 'react'
import { create } from 'react-test-renderer'
import Favorite from '../../../src/screens/Favorite'

jest.useFakeTimers()
jest.mock('@expo/vector-icons/FontAwesome5', () => 'Icon')
jest.mock('@react-native-community/async-storage', () => 'AsyncStorage')
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn()
    })
  }
})

describe('Favorite test', () => {
  it('render Favorite screen correctly', () => {
    const tree = create(<Favorite />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
