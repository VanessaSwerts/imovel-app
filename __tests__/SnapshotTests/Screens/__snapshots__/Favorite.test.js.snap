// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite snapshot test render Favorite screen correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "paddingHorizontal": "10%",
      "paddingVertical": "25%",
    }
  }
>
  <Icon
    color="#14213D"
    name="heart-broken"
    size={120}
  />
  <View>
    <Text
      style={
        Object {
          "color": "#333740",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Você ainda não possui imóveis favoritos
    </Text>
    <Text
      style={
        Object {
          "color": "#AAADB3",
          "fontSize": 18,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Vá para tela principal para visualizar os imóveis e favoritá-los!
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#19294d",
        "borderRadius": 24,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "opacity": 1,
        "width": 130,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Tela inicial
    </Text>
  </View>
</View>
`;
